src/lib.rs

```diff
@@ -294,9 +495,34 @@ impl<'input> Unit<'input> {
     }
 
     fn contains_address(&self, address: u64) -> bool {
-        self.ranges.iter().any(|range| address >= range.begin && address < range.end)
+        self.ranges().iter().any(|range| address >= range.begin && address < range.end)
     }
 
+    fn ranges(&self) -> &[gimli::Range] {
+        match *self {
+            Target::Function(ref u) => &u.ranges[..],
+            Target::Line(ref u) => &u.ranges[..],
+        }
+    }
+}
+
+struct FunctionUnit<'input> {
```

Comment:
Not sure it makes sense to include `Unit` in the names for these. The original `struct Unit` was the info for a compilation unit. Maybe `FunctionInfo/LineInfo` or something like that?